internal/commands/pull.go

```diff
@@ -1,34 +1,73 @@
 package commands
 
 import (
+	"context"
 	"fmt"
 	"os"
 
+	"github.com/docker/docker/api/types"
 	"github.com/docker/app/internal/cnab"
 	"github.com/docker/app/internal/packager"
 	"github.com/docker/app/internal/store"
 	"github.com/docker/cli/cli"
 	"github.com/docker/cli/cli/command"
 	"github.com/docker/cli/cli/config"
 	"github.com/docker/distribution/reference"
+	"github.com/docker/docker/pkg/jsonmessage"
+	"github.com/docker/docker/registry"
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
+type pullOptions struct {
+	serviceImages bool
+}
+
 func pullCmd(dockerCli command.Cli) *cobra.Command {
+	var opts pullOptions
+
 	cmd := &cobra.Command{
 		Use:     "pull APP_IMAGE",
 		Short:   "Pull an App image from a registry",
 		Example: `$ docker app pull myrepo/myapp:0.1.0`,
 		Args:    cli.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return runPull(dockerCli, args[0])
+			return runPull(dockerCli, opts, args[0])
 		},
 	}
+	cmd.Flags().BoolVar(&opts.serviceImages, "service-images", false, "Also pull down service images to this host's context")
 	return cmd
 }
 
-func runPull(dockerCli command.Cli, name string) error {
+func pullImage(ctx context.Context, cli command.Cli, image string) error {
+	ref, err := reference.ParseNormalizedNamed(image)
+	if err != nil {
+		return err
+	}
+
+	// Resolve the Repository name from fqn to RepositoryInfo
+	repoInfo, err := registry.ParseRepositoryInfo(ref)
+	if err != nil {
+		return err
+	}
+	authConfig := command.ResolveAuthConfig(ctx, cli, repoInfo.Index)
+	encodedAuth, err := command.EncodeAuthToBase64(authConfig)
+	if err != nil {
+		return err
+	}
+	options := types.ImagePullOptions{
+		RegistryAuth: encodedAuth,
+	}
+	responseBody, err := cli.Client().ImagePull(ctx, image, options)
+	if err != nil {
+		return err
+	}
+	defer responseBody.Close()
+
+	return jsonmessage.DisplayJSONMessagesStream(responseBody, cli.Out(), cli.Out().FD(), false, nil)
```

Comment:
As those multiple pulls may be really verbose, should we also add a `--quiet` flag to this command? WDYT @thaJeztah ?